onnxruntime/core/optimizer/constant_folding.cc

```diff
@@ -111,7 +111,10 @@ Status ConstantFolding::ApplyImpl(Graph& graph, bool& modified, int graph_level,
       }
 
       // Create execution frame for executing constant nodes.
-      OptimizerExecutionFrame::Info info({node}, constant_inputs, graph.ModelPath(), execution_provider_);
+      OptimizerExecutionFrame::Info info({node}, constant_inputs, graph.ModelPath(), execution_provider_,
+                                         [&graph](const std::string& name) -> bool {
+                                           return graph.IsSparseInitializer(name);
+                                         });
 
       std::vector<int> fetch_mlvalue_idxs;
```

Comment:
nit: can we move this lambda outside of the node loop and just provide a reference to it to the OptimizerExecutionFrame instead of copying it each time?

Reply:
Moved outside the loop.